### ISSUE TITLE
Not compatible with non-color terminal sessions

### ISSUE THREAD

**Issue by Stex** [zverok/yard-junk#25]
`Rainbow` [automatically disables itself](https://github.com/sickill/rainbow/blob/master/lib/rainbow.rb) if the terminal it runs in does not support colors for some reason. 

In this case, it uses its `Rainbow::NullPresenter` which is basically a no-op that only supports certain methods, including the [base colors](https://github.com/sickill/rainbow#rainbow-presenter-api).

In `YardJunk::Janitor::TextReporter`, the color `gray` is used which is not supported by said `NullPresenter`, resulting in a `NoMethodError` when being run on our CI system:

```
NoMethodError: undefined method `gray' for ", (%<duration>s to run)":Rainbow::NullPresenter
/usr/local/bundle/gems/yard-junk-0.0.5/lib/yard-junk/janitor/text_reporter.rb:19:in `<class:TextReporter>'
/usr/local/bundle/gems/yard-junk-0.0.5/lib/yard-junk/janitor/text_reporter.rb:10:in `<class:Janitor>'
/usr/local/bundle/gems/yard-junk-0.0.5/lib/yard-junk/janitor/text_reporter.rb:6:in `<module:YardJunk>'
/usr/local/bundle/gems/yard-junk-0.0.5/lib/yard-junk/janitor/text_reporter.rb:5:in `<top (required)>'
/usr/local/bundle/gems/yard-junk-0.0.5/lib/yard-junk/janitor.rb:85:in `require_relative'
/usr/local/bundle/gems/yard-junk-0.0.5/lib/yard-junk/janitor.rb:85:in `<top (required)>'
/usr/local/bundle/gems/yard-junk-0.0.5/lib/yard-junk.rb:7:in `require_relative'
/usr/local/bundle/gems/yard-junk-0.0.5/lib/yard-junk.rb:7:in `<top (required)>'
/usr/local/bundle/gems/yard-junk-0.0.5/lib/yard-junk/rake.rb:11:in `require_relative'
/usr/local/bundle/gems/yard-junk-0.0.5/lib/yard-junk/rake.rb:11:in `block in define_task'
```

I don't think this is a Rainbow issue as `gray` is not one of its base colors but rather an X11 color. The obvious solution would be to use a different color, but none of the base colors actually fits as nicely as gray.

Another solution might be to write a wrapper method inside the reporter which checks for rainbow availability and just skips it entirely if it's not enabled:

```ruby
class TextReporter
  ...

  private

  def rainbow(string, *commands)
    return string unless Rainbow.enabled
    commands.reduce(Rainbow(string)) { |r, c| r.send(c) }
  end
end

# rainbow('something', :green, :underline)
```

**Comment by zverok** [comment]
Thanks for reporter. Please try current master and tell me how it works for you.

**Comment by Stex** [comment]
@zverok Works like a charm, thanks a lot!
Looking forward to a new gem release including this change :)

**Comment by zverok** [comment]
@Stex Will do the release tomorrow! Currently too busy finishing my slides before my RubyKaigi talk :)

**Comment by zverok** [comment]
Released with 0.0.6 :tada:
